### PR TITLE
infoschema: print line number in parser slow log error for locating the wrong line

### PR DESCRIPTION
**Pull request by crazycs520**
### What problem does this PR solve? 
When TiDB parse `tidb-slow.log`, It may be error when parsing the wrong line, It is useful to prinout the wrong line number in the error message.

### What is changed and how it works?
Change parse slow log error message to printout the line number.

### Check List 

Tests 

 - Unit test

Code changes

 - Has exported function/method change

Side effects

Related changes

 - Need to cherry-pick to the release branch

Release note

 - Printout line number in parser slow log error for locating the wrong line

### CODE REVIEW

File: infoschema/slow_log.go [hunk 1]
```diff
@@ -282,7 +284,7 @@ func (st *slowQueryTuple) setFieldValue(tz *time.Location, field, value string)
 		st.sql = value
 	}
 	if err != nil {
-		return errors.Wrap(err, "parse slow log failed `"+field+"` error")
+		return errors.Wrap(err, "parse slow log on line "+strconv.FormatInt(int64(lineNum), 10)+" failed `"+field+"` error")
```
Comment (bb7133) [inline infoschema/slow_log.go]:
```suggestion
		return errors.Wrap(err, "parse slow log at line "+strconv.FormatInt(int64(lineNum), 10)+" failed `"+field+"` error")
```

File: infoschema/slow_log_test.go [hunk 2]
```diff
@@ -172,4 +172,16 @@ select * from t;`)
...
+	// Test parser error
+	slowLog = bytes.NewBufferString(
+		`# Time: 2019-05-12-11:23:29.614327491 +0800
+# Txn_start_ts: 405888132465033227#
+`)
+
+	scanner = bufio.NewReader(slowLog)
+	_, err = infoschema.ParseSlowLog(loc, scanner)
+	c.Assert(err, NotNil)
+	c.Assert(err.Error(), Equals, "parse slow log on line 2 failed `Txn_start_ts` error: strconv.ParseUint: parsing \"405888132465033227#\": invalid syntax")
```
Comment (bb7133) [inline infoschema/slow_log_test.go]: How about changing the message to:

```
"parse slow log on line 2 failed. Field: `Txn_start_ts`, error: strconv.ParseUint: parsing \"405888132465033227#\": invalid syntax"
```

**Comment by bb7133** [comment]
Please fix the CI @crazycs520

File: infoschema/slow_log_test.go (hunk 2 above)
Comment (crazycs520) [reply infoschema/slow_log_test.go]: Great, done.

File: infoschema/slow_log_test.go [hunk 3]
```diff
@@ -123,7 +123,7 @@ select * from t;
 	reader = bufio.NewReader(slowLog)
 	_, err = infoschema.ParseSlowLog(loc, reader)
 	c.Assert(err, NotNil)
-	c.Assert(err.Error(), Equals, "parse slow log failed `Succ` error: strconv.ParseBool: parsing \"abc\": invalid syntax")
+	c.Assert(err.Error(), Equals, "parse slow log at line 2 failed. Field: `Succ`, error: strconv.ParseBool: parsing \"abc\": invalid syntax")
```
Comment (djshow832) [inline infoschema/slow_log_test.go]: parse => Parse.
error: strconv.ParseBool: parsing \"abc\": invalid syntax => value: \"abc\"

File: infoschema/slow_log_test.go [hunk 4]
```diff
@@ -172,4 +172,16 @@ select * from t;`)
 	c.Assert(err, IsNil)
 	_, err = infoschema.ParseSlowLog(loc, scanner)
 	c.Assert(err, IsNil)
+
+	// Test parser error
```
Comment (AilinKid) [inline infoschema/slow_log_test.go]:
```suggestion
	// Test parser error.
```

File: infoschema/slow_log_test.go (hunk 3 above)
Comment (crazycs520) [reply infoschema/slow_log_test.go]: The error message is the original error message from `strconv.ParseBool`, I think to print it out is better.

Comment (AilinKid) [reply infoschema/slow_log_test.go]: @djshow832 you can try suggestion mode, which will be more clear.

**Comment by bb7133** [comment]
Please fix CI @crazycs520

**Pull request merged**